### PR TITLE
Add option to pass arbitrary validation code

### PR DESCRIPTION
**Pull request by lestrrat**
Meanwhile also convert most everything in `jwt.Validate()` into `jwt.Validator` based implementation

### CODE REVIEW

File: jwt/validate.go [hunk 1]
```diff
@@ -44,145 +44,282 @@ func timeClaim(t Token, clock Clock, c string) time.Time {
...
-			claim := o.Value().(claimValue)
-			claimValues[claim.name] = claim.value
+			validators = append(validators, v)
 		}
 	}
 
-	for c := range requiredMap {
-		if _, ok := t.Get(c); !ok {
-			return errors.Errorf(`required claim %s was not found`, c)
+	ctx := SetValidationCtxSkew(context.Background(), skew)
```
Comment (johejo) [inline jwt/validate.go:82]: Is it possible to propagate the parent context?

Comment (lestrrat) [reply jwt/validate.go:82]: hmmm, no, because `jwt.Validate()` does not take a context :/

Comment (johejo) [reply jwt/validate.go:82]: Is there any new API?

```go
jwt.Validate(
  jwt.ValidateWithContext(ctx),
)
```

Or make jwt.Validate depreacated
```go
jwt.ValidateWithContext(ctx, ...)
```

Comment (lestrrat) [reply jwt/validate.go:82]: Sure, it's _doable_ but why? is there a usecase?

Comment (johejo) [reply jwt/validate.go:82]: I considered some usecases for claim validation logic.
- RPC call to another server over network
- Add span with tracing framework such as OpenTelemetry

Comment (lestrrat) [reply jwt/validate.go:82]: Ah, the span, I can sympathize with...

Comment (lestrrat) [reply jwt/validate.go]: fixed in bb7f584

**Review by johejo — COMMENTED** [review]
LGTM

**Pull request merged**